android/src/main/java/com/rnmaps/maps/MapView.java

```diff
@@ -1375,7 +1375,13 @@ public void setIndoorActiveLevelIndex(int activeLevelIndex) {
   }
 
   private MapMarker getMarkerMap(Marker marker) {
-    MapMarker airMarker = markerMap.get(marker);
+    MapMarker airMarker = (MapMarker) marker.getTag();
```

Comment:
@MartinaMuscarella I would suggest creating a lightweight class in Android and call it MarkerTag
in it you can simply add two properties: viewId, markerId

then you can attach it to the marker as a tag

and where ever in the MapView getMarkerMap is called, replace it with a call to get the tagFirst as you did, and have a new manager.pushEvent function that takes the viewID directly, not just the viewObject